admin/load.php

```diff
@@ -503,3 +533,37 @@ function perflab_dismiss_wp_pointer_wrapper() {
 	check_ajax_referer( 'dismiss_pointer' );
 }
 add_action( 'wp_ajax_dismiss-wp-pointer', 'perflab_dismiss_wp_pointer_wrapper', 0 );
+
+/**
+ * Callback function hooked to admin_init to handle WPP inline standalone
+ * plugin actions such as activation and deactivation.
+ *
+ * @since n.e.x.t
+ * @return void
+ */
+function perflab_handle_inline_plugin_actions() {
+	if (
+		! empty( filter_input( INPUT_GET, 'wpp' ) ) &&
+		! empty( filter_input( INPUT_GET, 'action' ) ) &&
+		! empty( filter_input( INPUT_GET, 'plugin' ) ) &&
+		'1' === filter_input( INPUT_GET, 'wpp' ) &&
+		(
+			'deactivate' === filter_input( INPUT_GET, 'action' ) ||
+			'activate' === filter_input( INPUT_GET, 'action' )
+		)
+	) {
+		$referer = wp_get_referer();
+
+		switch ( filter_input( INPUT_GET, 'action' ) ) {
+			case 'activate':
+				activate_plugins( filter_input( INPUT_GET, 'plugin' ) );
+				break;
+			case 'deactivate':
+				deactivate_plugins( filter_input( INPUT_GET, 'plugin' ) );
+				break;
+		}
+		wp_safe_redirect( $referer );
+		exit;
+	}
+}
+add_action( 'admin_init', 'perflab_handle_inline_plugin_actions', 15, 0 );
```

Comment:
It would be a slightly cleaner solution to rely on the `admin_action_{$action}` hook instead, which is intended to process custom actions based on request parameters. Here's how that could work:
* Instead of having separate `wpp`, `action`, and `plugin` parameters, combine `wpp` and `action` to a single `action` parameter that could have one of the following values:
    * `perflab_activatee_plugin`
    * `perflab_deactivate_plugin`
    * Nitpick: Better use `perflab_` instead of `wpp_` since that's the prefix used so far.
* You can then implement two separate functions for the two actions (i.e. no more `switch` needed), and hook them into `admin_action_perflab_activate_plugin` and `admin_action_perflab_deactivate_plugin` respectively.
* In the function, you can still use the `plugin` parameter like you're doing here.

Reply:
@felixarntz this has been addressed in the following commits:

- https://github.com/WordPress/performance/pull/864/commits/bb58fc5982cebc657dc234313a52e2aa55ed51e7
- https://github.com/WordPress/performance/pull/864/commits/ddaf667b43e9519612466ed7a9269c55fb1dcd1c
- https://github.com/WordPress/performance/pull/864/commits/2583f3d0581bc348e1d8f0c6e3c4ff2dc511b2d9